awswrangler/_databases.py

```diff
@@ -181,6 +190,9 @@ def _iterate_results(
 ) -> Iterator[pd.DataFrame]:
     with con.cursor() as cursor:
         cursor.execute(*cursor_args)
+        if _oracledb_found:
+            decimal_dtypes = oracle.detect_oracle_decimal_datatype(cursor.description)
+        dtype = {**decimal_dtypes, **dtype} if decimal_dtypes and dtype is not None else decimal_dtypes
```

Comment:
I would think this line has to be inside the `if _oracledb_found:` statement. Otherwise, when oracle is not installed, `decimal_dtypes` variable would never exist and the following statement would fail. Surprised the linter did not catch this